step-grid-client/src/main/java/step/grid/client/GridObjectMapperResolver.java

```diff
@@ -28,14 +29,21 @@
 @Provider
 @Consumes(MediaType.APPLICATION_JSON)
 @Produces(MediaType.APPLICATION_JSON)
-public class ObjectMapperResolver implements ContextResolver<ObjectMapper> {
+public class GridObjectMapperResolver implements ContextResolver<ObjectMapper> {
 
     private final ObjectMapper mapper;
 
-    public ObjectMapperResolver() {
+    //HK2 still require a public no arg constructor, also used when no custom configuration is required (ex RemoteGridImpl)
+    public GridObjectMapperResolver() {
         mapper = new ObjectMapper();
     }
 
+    public GridObjectMapperResolver(int maxStringLength) {

```

Comment:
Can't we inject the GridClientConfiguration via annotation instead of using a constructor?

Reply:
@jeromecomte technically it's possible, but this oblige to define and bind such a configuration everywhere it is used.  This mapper provider is also used by the RemoteGrid and by the GridImpl (grid server). The server in particular does not have a client configuration and don't need such a customization.

Reply:
@david-stephan Get it. Not worth the effort. PR approved